packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ModelResource.java

```diff
@@ -244,4 +249,84 @@ public Response saveAsNewModel(
 		final ModelOperationCopy modelOperationCopy) {
 		return proxy.copyModel(modelOperationCopy);
 	}
+
+	@PUT
+	@Path("/generate-location-strata-model")
+	public Response generateLocationStrataModel (
+		@QueryParam("states") final String states) {
+		PetriNetModel petriNetModel = new PetriNetModel();
+		List<PetriNetState> petriNetStates;
+		List<PetriNetTransition> petriNetTransitions = new ArrayList<>();
+		String[] splitStateNames = states != null ? states.split(",") : new String[]{""};
+		if (splitStateNames.length < 1) {
+			return Response.status(Response.Status.BAD_REQUEST).build();
+		}
+		for (String stateName : splitStateNames) {
+			if (stateName.contains("_")) {
+				return Response.status(Response.Status.BAD_REQUEST).build();
+			}
+		}
+		petriNetStates = IntStream.range(0, splitStateNames.length).mapToObj(i -> {
+			PetriNetState petriNetState = new PetriNetState();
+			petriNetState.setId('L'+ Integer.toString(i + 1));
+			petriNetState.setName(splitStateNames[i]);
+			return petriNetState;
+		}).collect(Collectors.toList());
+		for (int i = 0; i < petriNetStates.size(); i++) {
+			for (int j = 0; j < petriNetStates.size(); j++) {
+				if (i != j) {
+					PetriNetTransition petriNetTransition = new PetriNetTransition();
+					petriNetTransition.setId('t' + Integer.toString(i + 1) + (j + 1));
+					petriNetTransition.setInput(Arrays.asList(petriNetStates.get(i).getId()));
+					petriNetTransition.setOutput(Arrays.asList(petriNetStates.get(j).getId()));
+					petriNetTransitions.add(petriNetTransition);
+				}
+			}
+		}
+		petriNetModel.setTransitions(petriNetTransitions);
+		petriNetModel.setStates(petriNetStates);
+		return Response
+			.status(Response.Status.OK)
+			.entity(petriNetModel)
+			.build();
+	}
+
+	@PUT
+	@Path("/generate-age-strata-model")
+	public Response generateAgeStrataModel(
+		@QueryParam("states") final String states) {
```

Comment:
Might be wrong here, but `@PUT` with `@QueryParam` seems weird, and I don't think it get around the the URL-max length problem. you need to use the Body if I am not mistaken.